poly/src/polynomial/univariate/dense.rs

```diff
@@ -538,24 +537,21 @@ impl<'a, F: Field> SubAssign<&'a DensePolynomial<F>> for DensePolynomial<F> {
     #[inline]
     fn sub_assign(&mut self, other: &'a DensePolynomial<F>) {
         if self.is_zero() {
-            self.coeffs.resize(other.coeffs.len(), F::zero());
-        } else if other.is_zero() {
-            return;
-        } else if self.degree() >= other.degree() {
-        } else {
-            // Add the necessary number of zero coefficients.
-            self.coeffs.resize(other.coeffs.len(), F::zero());
+            self.coeffs = other.coeffs.iter().map(|&x| -x).collect();
```

Comment:
The hope with the old way was that if self.coeffs has capacity large enough (e.g., it was the result of a previous operation that resulted in a zero), then we can avoid allocation.

Reply:
Oh yes you are right didn't notice that. Then there is no point to refactor this, this is already optimized, let us close :)